doc/whatsnew/fragments/8475.internal

```diff
@@ -0,0 +1,5 @@
+Following a deprecation period, ``is_typing_guard``, ``is_node_in_typing_guarded_import_block`` and
+``is_node_in_guarded_import_block``: from ``pylint.utils`` were removed: use a combination of
+``is_sys_guard`` and ``in_type_checking_block`` instead.
```

Comment:
We should probably create ``in_sys_guarded_block`` here, what do you think ?

Reply:
Do we have a need for this ourselves? If not I don't think so.

Reply:
We're probably doing something complicated to get the englobing if, and then check that it's a sys guard, but the design of ``in_type_checking_block`` is superior imo.  Also it's breaking change time and it would make those two functions consistent together, but ultimately you're right.

Reply:
>   ``is_sys_guard`` is still available, which was part of  ``is_node_in_guarded_import_block``.

So we do have the consistent equivalent.